workflow-controller/controller-service/src/test/java/io/hyscale/controller/directive/BuildSpecJsonHandlerTest.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.controller.directive;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.IOException;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.utils.WindowsUtil;
+import io.hyscale.controller.directive.impl.BuildSpecJsonHandler;
+import io.hyscale.controller.util.ServiceSpecTestUtil;
+import io.hyscale.servicespec.commons.fields.HyscaleSpecFields;
+import io.hyscale.servicespec.commons.model.service.Artifact;
+import io.hyscale.servicespec.commons.model.service.BuildSpec;
+import io.hyscale.servicespec.commons.model.service.ServiceSpec;
+
+public class BuildSpecJsonHandlerTest {
+
+    private static final String FILEPATH = "/servicespecs/buildSpec_update.hspec.yaml";
+
+    private static final String JSON_PATH = HyscaleSpecFields.getPath(HyscaleSpecFields.image,
+            HyscaleSpecFields.buildSpec);
+
+    private static BuildSpecJsonHandler buildSpecHandler = new BuildSpecJsonHandler();
```

Comment:
BuildSpecJsonHandler is a bean right ? Why are we creating a new object .
This will be error prone and we cannot rely on these test cases.